15.6.2 Release Prep

nhunzaker opened the pull request:
## Commits

- [x]  ChangeEventPlugin.js IE detection is broken on Chrome/Safari/FF etc https://github.com/facebook/react/issues/10032
- [x] No error when element with ref created in another elements constructor in dev environment #10025 
- [x] Treat columns property as a unitless number #10115
- [x] move augmentClass definition above SyntheticEvent Proxy replacement #10011
- [x] Use a closure to bind argument to callback in ReactErrorUtils #8363
- [x] Fix uncontrolled radios #10156
- [x] Add controlList to DOM property whitelist #9940

## Documentation

- [x] Changelog 
- [x] Documentation updates

## Misc

- [x] Update lockfile
- [x] Version bump

nhunzaker commented:
@aweary @flarnie I think this is good to go, I think the only outstanding tasks beyond deployment are to update the static markup for the website and ship to Bower.

But this PR cherry picks the commits we want (aside from #10025, which we need to figure out) and sets up everything else.

gaearon commented:
Is anything blocking https://github.com/facebook/react/pull/10025? I left a comment there: https://github.com/facebook/react/pull/10025#issuecomment-321973500. I think if that change is made (and code only touches Stack rather than Fiber) then the change is good and we should get it in.

colebowl commented:
@gaearon - It looks like @iansu has done the rebase in issue #10025 and is passing all tests.

iansu commented:
I didn't think to comment in this thread. I have reverted the Fiber changes and updated my tests. Everything is passing.

I've been trying not to bug Dan because I think he's on vacation. 😀

nhunzaker commented:
Awesome. @iansu thanks!

nhunzaker commented on src/renderers/shared/stack/reconciler/__tests__/refs-test.js:
```diff
@@ -283,3 +284,78 @@ describe('ref swapping', () => {
+      __DEV__ = originalDev;
+      process.env.NODE_ENV = originalEnv;
+    }
+  });
```
Controlling for `process.env.NODE_ENV` is new here. Otherwise, I was unable to get the invariant to minify the warning.

nhunzaker replied:
Okay, I've done that in 26722ce and added a TODO to follow up with a similar change on master.

nhunzaker commented:
Everything checked off. I believe this is good to go.

gaearon commented on src/renderers/shared/stack/reconciler/__tests__/refs-test.js:
```diff
@@ -283,3 +284,78 @@ describe('ref swapping', () => {
+      __DEV__ = originalDev;
+      process.env.NODE_ENV = originalEnv;
+    }
+  });
```
The way it is written here will break other things in non-trivial way. Please check ReactDOMProduction for example of test that overrides process.env.

gaearon replied:
In fact it would be great if you could send a PR to master that fixes DEV override to work same way as in ReactDOMProduction-test.

nhunzaker replied:
It will be done.

nhunzaker commented:
It will be done.

iansu commented on CHANGELOG.md:
```diff
@@ -1,3 +1,15 @@
+* Prevent event handlers from receiving extra argument (dev only) ([@aweary](https://github.com/aweary) in [#10115](https://github.com/facebook/react/pull/8363)
+* Fix cases where onChange would not fire with defaultChecked on radio inputs ([@jquense](https://github.com/jquense) in [#10156](https://github.com/facebook/react/pull/10156))
+* Add support for controlList attribute to DOM property whitelist ([@nhunzaker](https://github.com/nhunzaker) in [#9940](https://github.com/facebook/react/pull/9940))
+* Fix a bug where creating an element with a ref in a constructor threw an error in dev mode ([@iansu](https://github.com/iansu) in [#10025](https://github.com/facebook/react/pull/10025))
```
This description is wrong. The bug was that creating an element with a ref in a constructor did **NOT** throw an error in dev mode.

nhunzaker replied:
🙈 Thank you.

nhunzaker replied:
Got it in da62f42

colebowl commented:
@aweary - Would you be able to complete the requested review? 

@iansu - Are there any other steps we need to complete before this can be merged?

ssilve1989 commented:
Any update when this will be available? I'm looking to get the fixes for uncontrolled radio buttons.

manikandasankar commented:
Any update when this will be available?
I am struck with this issue [ChangeEventPlugin.js IE detection is broken](https://github.com/facebook/react/pull/10032)

colebowl commented:
Hi @gaearon / @aweary / @nhunzaker,

I know you guys aren't actively working on this branch as you're working on on the next version but I was wondering if it would be possible to get this wrapped up and merged? There are several folks waiting on this to fix issues in their app (including us).

If there is any more action that needs to be taken, I'd be more than happy take the lead to get this done.

Thanks!

nhunzaker commented:
@colebowl Thanks for the prod, and I am sorry it was necessary.

I've updated the ref test to use the recommended code and I've updated the timestamps on the changelog and blog post.

This should be good to go, but I'll be out of pocket the rest of the day. I'm happy to review any updates if we need to change anything. I'll circle back later tonight.

gaearon commented:
We'll need to get the LICENSE change rolled here. Plan to get it out tomorrow.

sophiebits commented:
I can just do it after merging this. Probably easiest that way.
